e2e/machine_migration_test.go

```diff
@@ -237,6 +256,179 @@ var _ = Describe("[sig-cluster-lifecycle][OCPFeatureGate:MachineAPIMigration] Ma
 			})
 		})
 	})
+
+	var _ = Describe("Update MAPI Machine", Ordered, func() {
+		Context("with spec.authoritativeAPI modification", func() {
+			It("should allow modification of authoritativeAPI from ClusterAPI to MachineAPI", func() {
+				By("Attempting to modify authoritativeAPI from ClusterAPI to MachineAPI")
+
+				// Create a new machine for this test to avoid affecting other tests
+				testMachineName := "machine-auth-change-83955"
+				testMachine := createMAPIMachineWithAuthority(ctx, cl, testMachineName, machinev1beta1.MachineAuthorityClusterAPI)
+
+				_, err := capiframework.GetMachine(cl, testMachineName, capiframework.CAPINamespace)
+				Expect(err).ToNot(HaveOccurred(), "CAPI Machine should be created")
+				// Ensure any pending updates are processed
+				Eventually(func() error {
+					// Get the latest version again right before update
+					currentMachine, err := mapiframework.GetMachine(cl, testMachineName)
+					if err != nil {
+						return err
+					}
+
+					updatedMachine := currentMachine.DeepCopy()
+					updatedMachine.Spec.AuthoritativeAPI = machinev1beta1.MachineAuthorityMachineAPI
+
+					err = cl.Update(ctx, updatedMachine)
+					return err
+				}, capiframework.WaitMedium, capiframework.RetryMedium).Should(Succeed(), "Should allow modification of authoritativeAPI")
+
+				// Verify the machine still exists after the change (don't wait for specific status)
+				_, err = mapiframework.GetMachine(cl, testMachineName)
+				Expect(err).ToNot(HaveOccurred(), "Machine should still exist after authoritativeAPI change")
```

Comment:
Can we use Komega here? 
```suggestion
				Eventually(k.Update(testMachine, func(){
				testMachine.Spec.AuthoritativeAPI = machinev1beta1.MachineAuthorityMachineAPI
				}), capiframework.WaitMedium, capiframework.RetryMedium).Should(Succeed(), "Should allow modification of authoritativeAPI")
```

Reply:
Thank you Theo using this , need to updated `machinev1beta1.MachineAuthorityMachineAPI` to `machinev1beta1.MachineAuthorityClusterAPI`